Quote row type field names

martint opened the pull request:
Since row type field names can match reserved SQL keywords, names
need to be quoted when rendering them.

Fixes https://github.com/prestosql/presto/issues/1962

martint committed:
Use getDisplayName instead of TypeSignature in error message

martint committed:
Quote identifier name during row subscript desugar

Names can be reserved SQL keywords, so we need to quote them
to avoid issues when deserializing plans on workers.

martint committed:
Use display name for Type's toString

martint committed:
Remove use of TypeSignature from OperatorNotFoundException

kokosing commented on presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java:
```diff
@@ -319,6 +319,11 @@ public void testRowSubscript()
         assertQuery("SELECT n.name, r.name FROM nation n JOIN region r ON ROW (n.name, n.regionkey)[2] = ROW (r.name, r.regionkey)[2] ORDER BY n.name LIMIT 1", "VALUES ('ALGERIA', 'AFRICA')");
+
+        // Subscript over field named after reserved keyword
+        assertQuery(
```
I think we should have a dedicated test for planner (`TestPlanner`?). There is no point to run this test for all the connectors.

dain replied:
IIRC the issue occurs when distributing plan fragments to workers, so this must run in distributed mode.

kokosing commented on presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java:
```diff
@@ -446,6 +446,16 @@ public void testRowFieldAccessorInJoin()
     }
 
+    @Test
+    public void testRowTypeWithReservedKeyword()
```
same comment

martint replied:
Agreed, but see the comment below:

```
 // This test is here because it only reproduces the issue (https://github.com/prestosql/presto/issues/1962)
// when running in distributed mode
```

findepi replied:
so it seems we  need dedicated test for planner (TestPlanner?), using distributed tpch runner.

martint replied:
That's a great idea (for a separate change). For now, I'm moving it to TestTpchDistributedQueries with a comment.

martint committed:
Quote field names in TypeSignature

TypeSignature is used for encoding the identity of a type, which
is then parsed (for now) using the SQL parser. Since some fields
names might collide with SQL reserved words, we need to quote them.

findepi commented:
> IIRC the issue occurs when distributing plan fragments to workers, so this must run in distributed mode.

`TestTpchDistributedQueries`?

it already is used to test distribution-related things, eg `testTooManyStages`
